packages/esm-patient-chart-app/src/visit/visit-form/visit-type-overview.component.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import Search from 'carbon-components-react/es/components/Search';
+import RadioButtonGroup from 'carbon-components-react/es/components/RadioButtonGroup';
+import RadioButton from 'carbon-components-react/es/components/RadioButton';
+import { useTranslation } from 'react-i18next';
+import styles from './visit-type-overview.component.scss';
+import debounce from 'lodash-es/debounce';
+import { usePagination, useVisitTypes, VisitType } from '@openmrs/esm-framework';
+import isEmpty from 'lodash-es/isEmpty';
+import { PatientChartPagination } from '../../../../esm-patient-common-lib/src';
+
+interface VisitTypeOverviewProps {
+  isTablet: boolean;
+  onChange: (event) => void;
+}
+
+const VisitTypeOverview: React.FC<VisitTypeOverviewProps> = ({ isTablet, onChange }) => {
+  const { t } = useTranslation();
+  const visitTypes = useVisitTypes();
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [searchResults, setSearchResults] = useState<Array<VisitType>>([]);
+
+  useEffect(() => {
+    if (!isEmpty(searchTerm)) {
+      const results = visitTypes.filter(
+        (visitType) => visitType.display.toLowerCase().search(searchTerm.toLowerCase()) !== -1,
+      );
+      setSearchResults(results);
+    } else {
+      setSearchResults(visitTypes);
+    }
+  }, [searchTerm, visitTypes]);
+
+  const handleSearch = useMemo(() => debounce((searchTerm) => setSearchTerm(searchTerm), 300), []);
```

Comment:
```suggestion
  const handleSearch = () => debounce((searchTerm) => setSearchTerm(searchTerm), 300);
```

I don't thiiiiiiink there's any reason to use `useMemo` here

Reply:
Important: The suggestion is wrong. `useMemo` accepts a callback to derive a value, so the suggested change would always call `debounce` and create a new debounced handler from it.

Instead it is:

```js
const handleSearch = debounce((searchTerm) => setSearchTerm(searchTerm), 300);
```

Presumably, the `useMemo` is indeed not really useful / worth it in here, but it all depends on what calling `debounce` does under the hood.

Reply:
Whoops! That's right. That's also what we have in [patient search](https://github.com/openmrs/openmrs-esm-patient-management/blob/c150ccd37f52bbee28c9063ac5accd82b0a0f4c9/packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx#L62). If we need to optimize I think we'd [do so with `useCallback`](https://stackoverflow.com/a/55616626/1464495).

Reply:
Hm in general yes, but in this particular case I'd argue no. If you'd use `useCallback(debounce(...))` then you'd already walk through the potentially expensive function call. Even worse - whatever you created there gets most likely thrown away unless dependencies changed. In this case `useMemo` would be preferred, but as written this depends on the cost of calling `debounce` (or phrased differently "the cost of constructing a debounced function").